src/library/searchquery.h

```diff
@@ -57,6 +57,7 @@ class AndNode : public GroupNode {
 class NotNode : public QueryNode {
   public:
     explicit NotNode(QueryNode* pNode);
+    ~NotNode();
```

Comment:
Only the destructor in the base class needs to be virtual. The keyword 'virtual' is redundant in derived classes. For functions we will use 'override' in derived classes (C++11), but there is nothing similar for the destructor.